sdk/storage/azure-storage-blobs/inc/azure/storage/blobs.hpp

```diff
@@ -17,3 +17,4 @@
 #include "azure/storage/blobs/block_blob_client.hpp"
 #include "azure/storage/blobs/dll_import_export.hpp"
 #include "azure/storage/blobs/page_blob_client.hpp"
+#include "azure/storage/common/storage_exception.hpp"
```

Comment:
Why do we need this here, especially now?

We should avoid exposing headers from other packages from within the simplified main.hpp header for the package.

For example: blobs.hpp shouldn't contain headers from storage common, and only contain all the main headers we want customers to include from blobs. If those blobs header transitively include other headers, that's fine, but they shouldn't be in the main simplified header.

Reply:
In the old protocol layer code, function definitions were in .hpp file, so `StorageException` was indirectly exposed to customers. Customers could use this it without including the header.

But in the new protocol layer code, function definitions were moved to .cpp file, thus customers don't have access to the exception unless they explicitly include `storage_exception.hpp`. Unfortunately, storage exception is something that's being used by a lot of customers, and it's not easy for them to figure out that they should include another header after upgrading to a new version for the existing code to keep working. This is like a breaking change. So I added `storage_exception.hpp` to `blobs.hpp`, `queues.hpp` and `datalake.hpp` as a mitigation.

Reply:
I know this is not ideal. But I cannot come up with a better solution.